File: examples/ruby_on_rails_ruby_graphql/app/models/reaction.rb
# frozen_string_literal: true

# == Schema Information
#
# Table name: reactions
#
#  id         :uuid             not null, primary key
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  emotion    :string
#  post_id    :uuid             not null
#  user_id    :uuid             not null
#
class Reaction < ApplicationRecord
  belongs_to :post
  belongs_to :user

  enum_a emotion: [:like, :dislike, :love, :laugh, :cry, :wow]
end

File: examples/go/graph/model/reaction.go
package model

import "github.com/google/uuid"

type Reaction struct {
	ID        uuid.UUID       `json:"id" gorm:"type:uuid;default:gen_random_uuid()"`
	CreatedAt ISO8601DateTime `json:"createdAt" gorm:"not null"`
	UpdatedAt ISO8601DateTime `json:"updatedAt" gorm:"not null"`
	PostId    uuid.UUID       `json:"-" gorm:"not null"`
	Post      *Post           `json:"post"`
	UserId    uuid.UUID       `json:"-" gorm:"not null"`
	User      *User           `json:"user"`
	Emotion   Emotion         `json:"emotion" gorm:"not null"`
}

// TODO: Create and use database ENUM for `Emotion`
